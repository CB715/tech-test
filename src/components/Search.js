import React, { useState } from 'react';
import PropTypes from 'prop-types'

import '../styles/search.css';
import getImages from '../requests/getImages';

const Search = ({ setSearchResults }) => {
    const [value, setValue] = useState();
    
    const handleSubmit = async (event) => {
        event.preventDefault();
        setSearchResults(await getImages(value));
    };

    return (
    <div>
      <form className='search-form' onSubmit={handleSubmit}>
        
        <input 
          className='search-form__input-box' 
          type='text' 
          placeholder='Search' 
          onChange={(e) => setValue(e.target.value)}
        />
    
        <button className='search-form__button' type='submit' data-testid='button'>
          Search
        </button>
      </form>
    </div> 
    );
};

Search.propTypes = {
  setSearchResults: PropTypes.func,
};

Search.defaultProps = {
  setSearchResults: () => [],
}

export default Search;
