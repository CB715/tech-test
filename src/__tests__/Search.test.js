import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react'
import Search from "../components/Search";

describe('Search', () => {
const validProps = {
  setSearchResults: jest.fn()
}

    it('renders correctly', () => {
      const { asFragment } = render(<Search setSearchResults={validProps.setSearchResults} />);
      
      expect(asFragment()).toMatchSnapshot();
  });

  it('calls function when button fired', async() => {
    render(<Search setSearchResults={validProps.setSearchResults} />);
    const button = screen.getByTestId('button')
    await fireEvent.click(button);

    expect(validProps.setSearchResults).toHaveBeenCalled();
  });

  it('displays button', () => {
    render(<Search setSearchResults={validProps.setSearchResults} /> );
    const button = screen.getAllByRole('button');

    expect(button).toHaveLength(1);
  });
});