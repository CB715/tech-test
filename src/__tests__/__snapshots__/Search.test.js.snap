// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<DocumentFragment>
  <div>
    <form
      class="search-form"
    >
      <input
        class="search-form__input-box"
        placeholder="Search"
        type="text"
      />
      <button
        class="search-form__button"
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
</DocumentFragment>
`;
